### PR TITLE
Kubernetes entrypoint s3 override

### PR DESCRIPTION
**Pull request by alexisdondon**
allow override of entrypoint kubernetes job when METAFLOW_S3_ENDPOINT_URL is set in conf.
Test on a minio.
Not test on aws and @kubernetes need s3 datastore.
Read it carefully but not a lot of code

### CODE REVIEW

File: metaflow/metaflow_environment.py [hunk 1]
```diff
@@ -89,10 +89,10 @@ def get_package_commands(self, code_package_url):
             "mkdir .metaflow",  # mute local datastore creation log
             "i=0; while [ $i -le 5 ]; do "
             "mflog 'Downloading code package...'; "
-            "%s -m awscli s3 cp %s job.tar >/dev/null && \
+            "%s -m awscli %s s3 cp %s job.tar >/dev/null && \
                         mflog 'Code package downloaded.' && break; "
             "sleep 10; i=$((i+1)); "
-            "done" % (self._python(), code_package_url),
+            "done" % (self._python(),"--endpoint-url "+s3_endpoint_url if s3_endpoint_url is not None else "" , code_package_url),
```
[the remark this answers is not in the document]
Comment (alexisdondon) [reply metaflow/metaflow_environment.py:95]: yes the value of the parameter s3_endpoint_url is indeed pull from METAFLOW_S3_ENDPOINT_URL.
I followed the structure of the code where all env variable seems to be injected in kubernetes.py.

[the remark this answers is not in the document]
Comment (alexisdondon) [reply metaflow/metaflow_environment.py:95]: to finally made the helloworld job completely run on on premise kubernetes + minio i have to add with this version of metaflow to add   METAFLOW_S3_ENDPOINT_URL in my secret or adding this code to directly inject in env variable of the container.
```
     #add METAFLOW_S3_ENDPOINT_URL in env variable if specified for step execution
        if S3_ENDPOINT_URL is not None:
            job.environment_variable("METAFLOW_S3_ENDPOINT_URL",S3_ENDPOINT_URL)
```
i don't know what u prefer.

In summary there is two needs:
* allowing awscli to download the package no escape possible from adding a conditional --endpoint-url (this is my PR)
* adding METAFLOW_S3_ENDPOINT_URL in env variable in order to run the step in the container i can update my PR adding the code to conditionnally add it or you can say it's up to the user to specify this in the secrets

[the remark this answers is not in the document]
Comment (alexisdondon) [reply metaflow/metaflow_environment.py]: Don't forget to squash commit if you want to merge as i make a mess wanting to amend a commit :-)

**Comment by alexisdondon** [comment]
Is it expected something from me to make this PR move forward?

**Comment by savingoyal** [comment]
@oavdeev can you take a look at this PR?

**Comment by oavdeev** [comment]
I think some changes need to be made to AWS Batch plugin too (this is from our internal tests):
```
Traceback (most recent call last):
File ".../outerbounds/integration-test-suite-aws-batch/metaflow/plugins/aws/batch/batch_cli.py", line 296, in step
num_parallel=num_parallel,
File ".../outerbounds/integration-test-suite-aws-batch/metaflow/plugins/aws/batch/batch.py", line 318, in launch_job
num_parallel=num_parallel,
File ".../outerbounds/integration-test-suite-aws-batch/metaflow/plugins/aws/batch/batch.py", line 200, in create_job
self.environment, code_package_url, step_name, [step_cli], task_spec
File ".../outerbounds/integration-test-suite-aws-batch/metaflow/plugins/aws/batch/batch.py", line 62, in _command
init_cmds = environment.get_package_commands(code_package_url)
TypeError: get_package_commands() missing 1 required positional argument: 's3_endpoint_url'
```

**Comment by savingoyal** [comment]
#1045 introduces this functionality

**Pull request closed without merging**